# Session 7 — One-sample hypothesis tests

Revision ID: S07-r1  
Estimated time: 75–90 minutes

This notebook uses a reproducible synthetic software-deployment benchmark dataset. It is a teaching scenario, not production evidence. Download `deployment_benchmarks.csv` from the Session 7 page and keep it next to this notebook (or upload it in Colab).

## Load and identify the data

One row is one simulated benchmark run. `design` distinguishes independent cases from repeated measurements; in the paired cohort, the same `case_id` appears under both versions. Do not pool the cohorts for a single test.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats

benchmark = pd.read_csv("deployment_benchmarks.csv")
independent = benchmark.loc[benchmark["design"] == "independent"].copy()
paired = benchmark.loc[benchmark["design"] == "paired"].copy()
print(f"Rows: {len(benchmark)}; columns: {benchmark.shape[1]}")
display(benchmark.groupby(["design", "version"])["response_time_ms"].agg(["count", "mean", "std"]).round(2))

Rows: 280; columns: 7


count    mean    std
design      version                        
independent baseline      80  240.48  31.19
            optimized     80  231.76  30.02
paired      baseline      60  237.83  20.35
            optimized     60  221.61  23.59

## S07-E1 — Run and report a one-sample t test

Select the optimized runs in the independent cohort. Test H0: the population mean response time is 240 ms against a two-sided alternative. Report n, the sample mean, SD, t statistic, df, p-value, 95% confidence interval for the mean, and Cohen's d. State the unit.

In [2]:
# TODO: select the sample, run the two-sided test, and calculate the 95% t interval and Cohen's d.
result = None

### Interpretation

Write your contextual interpretation here. Include the units, design, and a limitation where requested.

## S07-E2 — Separate statistical evidence from practical importance

Using alpha=0.05, state the test decision and interpret the estimate, confidence interval, and standardized effect in context. Explain why a small p-value would not by itself establish a practically important or causal improvement. A one-sided alternative is defensible only if it was selected before inspecting these data.

In [3]:
# TODO: compare the p-value with alpha and write a contextual interpretation.
interpretation = None

### Interpretation

Write your contextual interpretation here. Include the units, design, and a limitation where requested.

## S07-E3 — Use simulation to distinguish Type I error and power

Simulate 2,000 two-sided one-sample t tests with n=20 and sigma=30 under (a) a true mean of 240 ms and (b) a true mean of 230 ms. At alpha=0.05, estimate the rejection rate in each condition. Explain which rate estimates Type I error and which estimates power, and why power is not guaranteed to be high.

In [4]:
# TODO: simulate both conditions and estimate rejection rates at alpha=.05.
type_one_rate = None
power_estimate = None

### Interpretation

Write your contextual interpretation here. Include the units, design, and a limitation where requested.

## Before submitting

- Run every cell from a fresh kernel.
- Identify the observational unit and comparison design.
- Report estimates and uncertainty with units.
- Separate statistical evidence from practical importance and causality.
- Keep claims within this synthetic dataset's scope.